pkg/tfvars/vsphere/vsphere.go

```diff
@@ -0,0 +1,55 @@
+package vsphere
+
+import (
+	"encoding/json"
+
+	vsphereprovider "sigs.k8s.io/cluster-api-provider-vsphere/api/v1alpha3"
+	"sigs.k8s.io/cluster-api-provider-vsphere/api/v1alpha3/cloudprovider"
+)
+
+type config struct {
+	VSphereURL      string `json:"vsphere_url"`
+	VSphereUsername string `json:"vsphere_username"`
+	VSpherePassword string `json:"vsphere_password"`
+	MemoryMiB       int64  `json:"control_plane_memory_mib"`
+	DiskGiB         int32  `json:"control_plane_disk_gib"`
+	NumCPUs         int32  `json:"control_plane_num_cpus"`
+	Cluster         string `json:"vsphere_cluster"`
+	Datacenter      string `json:"vsphere_datacenter"`
+	Datastore       string `json:"vsphere_datastore"`
+	ResourcePool    string `json:"vsphere_resource_pool"`
+	Folder          string `json:"vsphere_folder"`
+	Network         string `json:"vsphere_network"`
+	Template        string `json:"vsphere_template"`
+}
+
+// TFVarsSources contains the parameters to be converted into Terraform variables
+type TFVarsSources struct {
+	CloudProviderConfiguration cloudprovider.Config
```

Comment:
can probably be replaced by individual configuration which will be available for install-config.